### ISSUE TITLE
Artefacts with `LMSDiscreteScheduler` at certain step counts

### ISSUE THREAD

**Issue by psychedelicious** [huggingface/diffusers#3598]
### Describe the bug

`LMSDiscreteScheduler` often produces images with artefacts.

This has been an issue for some time (several weeks at least), and I'm surprised there aren't existing reports. Perhaps the reason is that `LMSDiscrete` is not a particularly popular scheduler.

Here are my notes:
- as step count increases, artefacting decreases - but I've run into this many times with step counts over 100
- artefacts look similar to decoded noise
- artefacts are often around regions you'd identify as a "face" or "eyes" - the examples in this issue show the artefacts in other places as well, but typically they are most prevalent in faces
- artefacts occur, for a given prompt and seed, intermittently across various step counts
- artefacts are deterministic; the same settings produce the same image
- `xformers` and VAE selection do not appear to be a factor
- some models are affected far more than others - the base SD1.5 and SD2 models are the least affected, but it still happens with them
- highly dependent on prompt - the examples here use the prompt in the repro section

Examples with `XpucT/Deliberate` and `LMSDiscreteScheduler`:

*Step 74*
![XpucT_Deliberate--74--1337926281](https://github.com/huggingface/diffusers/assets/4822129/54514e49-4d1f-45a7-a359-16dcb03af85f)

*Step 75*
![XpucT_Deliberate--75--1337926281](https://github.com/huggingface/diffusers/assets/4822129/78d70e8b-e34b-44fa-a21a-d40e4a323a58)

*Step 76*
![XpucT_Deliberate--76--1337926281](https://github.com/huggingface/diffusers/assets/4822129/d0becea2-0765-4254-8a18-797513990d5a)

*Step 77*
![XpucT_Deliberate--77--1337926281](https://github.com/huggingface/diffusers/assets/4822129/505d86cc-e58c-45d3-8ed4-d77a65a24de5)

*Step 78*
![XpucT_Deliberate--78--1337926281](https://github.com/huggingface/diffusers/assets/4822129/ed45eee5-9009-4224-ab18-baf27cd671b5)

*Step 79*
![XpucT_Deliberate--79--1337926281](https://github.com/huggingface/diffusers/assets/4822129/07e4dfce-17b7-47c8-b1f3-abd286c3104a)

*Step 80*
![XpucT_Deliberate--80--1337926281](https://github.com/huggingface/diffusers/assets/4822129/585401bc-d82c-44b0-b7fb-c787af005e0a)

### Reproduction

Here's a simple script to reproduce the issue, using `deliberate`, `sd2-base` and `sd2` for models and `lms`, `ddpm` and `euler` discrete schedulers. It makes images for 1 to 100 steps. 

```python
import torch
from diffusers import StableDiffusionPipeline, LMSDiscreteScheduler, AutoencoderKL, DDPMScheduler, EulerDiscreteScheduler

for m in ["XpucT/Deliberate", "stabilityai/stable-diffusion-2-base","stabilityai/stable-diffusion-2"]:
    schedulers = {
        "ddpm": DDPMScheduler.from_pretrained(m, subfolder="scheduler"),
        "lms": LMSDiscreteScheduler.from_pretrained(m, subfolder="scheduler"),
        "euler": EulerDiscreteScheduler.from_pretrained(m, subfolder="scheduler"),
    }
    vae = AutoencoderKL.from_pretrained("stabilityai/sd-vae-ft-mse", torch_dtype=torch.float16).to("cuda")

    for d in schedulers.items():
        pipeline = StableDiffusionPipeline.from_pretrained(m, vae=vae, scheduler=d[1], torch_dtype=torch.float16).to("cuda")
        for i in range(1, 100):
            for s in [1337926281]:
                filename = f"{m.replace('/', '_')}--{d[0]}--{i}--{s}.png"
                generator = torch.Generator(device="cuda").manual_seed(s)
                pipeline(prompt='elephant playing with a baby elephant in a zoo ', num_inference_steps=i, generator=generator).images[0].save(filename)
```

### System Info

My system:
- diffusers 0.16.1
- torch 2.0.0+cu118
- Ubuntu 22.04
- RTX 4090

This occurs on a wide range of system configurations, including cloud servers with big fancy GPUs.

**Comment by hipsterusername** [comment]
Related to [#3596 LMS Comparison](https://github.com/huggingface/diffusers/issues/3596)

**Comment by patrickvonplaten** [comment]
Hmm, I wonder whether this might simply be a problem with the scehduler - @anton-l any ideas from your side since you've added the scheduler? 

Overall, I don't recommend using the LMSDiscreteScheduler. https://huggingface.co/docs/diffusers/api/schedulers/multistep_dpm_solver#diffusers.DPMSolverMultistepScheduler is much better IMO

**Comment by patrickvonplaten** [comment]
@psychedelicious does it only happen with the `LMSDiscreteScheduler` or also with `EulerDiscreteScheduler`? Also we should probably double check if the same happens with  https://github.com/huggingface/diffusers/blob/07ef4855cd2b2fe9d72bc0479f15959333f11068/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_k_diffusion.py#L48

E.g.:

```py
from diffusers import StableDiffusionKDiffusionPipeline

pipeline = StableDiffusionKDiffusionPipeline.from_pretrained("XpucT/Deliberate")
pipeline.set_scheduler("sample_lms")

# .. test with this pipeline if the same error persists
```

If it doesn't happen with the original LMS code then there is def a problem in `diffusers`

**Comment by psychedelicious** [comment]
I'll do some testing with the `k_diffusion` code you've referenced @patrickvonplaten in a day or two. Thanks for the guidance.

**Comment by StAlKeR7779** [comment]
Tested this with `k_diffusion` - artifacts persists, so looks that this is fine.
Looks like our old generation pipeline somehow works better with lms, even if this was not right as generated image differs from diffusers or a1111 output.
Steps 80-86(diffusers - k_diffusion)
![XpucT_Deliberate-combine--sample_lms--80--1337926281](https://github.com/huggingface/diffusers/assets/7768370/691186d3-9528-4c01-9f35-56645df04de3)
![XpucT_Deliberate-combine--sample_lms--81--1337926281](https://github.com/huggingface/diffusers/assets/7768370/12960cda-db0c-4c1b-810a-94319aa01edf)
![XpucT_Deliberate-combine--sample_lms--82--1337926281](https://github.com/huggingface/diffusers/assets/7768370/642ba565-84af-4a10-ab24-491bf43feb1b)
![XpucT_Deliberate-combine--sample_lms--83--1337926281](https://github.com/huggingface/diffusers/assets/7768370/68d295f8-36e2-4c13-a1da-0d485ecbd239)
![XpucT_Deliberate-combine--sample_lms--84--1337926281](https://github.com/huggingface/diffusers/assets/7768370/36ee1658-7196-42f4-9a6b-44a50270f0d9)
![XpucT_Deliberate-combine--sample_lms--85--1337926281](https://github.com/huggingface/diffusers/assets/7768370/2b8ee30b-83df-4b39-a08a-a6ad6869d76d)
![XpucT_Deliberate-combine--sample_lms--86--1337926281](https://github.com/huggingface/diffusers/assets/7768370/27a568f2-8e6f-46e8-8f8c-f23843c08f51)
